# CMPT 423/820 
## Assignment 2 Question 3
* Austin Heinrich
* 11177796
* awh055

Use the A2Q3 dataset to compare three classifiers:

1) K-Nearest Neibours Classifier. Remember that you'll have to choose *K*.

2) Naive Bayes Classifier

3) Decision Tree Classifier

### K-Neighbors Classifier

In [1]:
import pandas as pd
from sklearn import neighbors
from sklearn.model_selection import train_test_split
from sklearn.metrics import f1_score 

dataframe = pd.read_csv('a2q3.csv', header=None, 
                        names=['Class','I','II','III','IV','V','VI','VII','VIII','IX','X','XI','XII','XIII'],
                        index_col=False)
features = dataframe.drop(columns=['Class'])
labels = dataframe['Class']

# split the dataframe into a training set and a testing set
tpropn = 0.5
features_train, features_test, labels_train, labels_test = train_test_split(features, labels, test_size=tpropn)

K = 14

clf = neighbors.KNeighborsClassifier(K)
clf.fit(features_train, labels_train)
predict_kn = clf.predict(features_test)
print("F1 Score:", f1_score(labels_test, predict_kn, average='weighted'))

F1 Score: 0.7021963822807591


### Gaussian Naive Bayes Classifier

In [2]:
from sklearn.naive_bayes import GaussianNB

clf = GaussianNB()
clf.fit(features_train, labels_train)
predict_gauss = clf.predict(features_test)

print("F1 Score:", f1_score(labels_test, predict_gauss, average='weighted'))

F1 Score: 0.9661538745149367


### Decision Tree Classifier

In [3]:
from sklearn.tree import DecisionTreeClassifier
from sklearn.datasets import load_iris

depth = 10

clf = DecisionTreeClassifier(max_depth=depth)
clf.fit(features_train, labels_train)
predict_dt = clf.predict(features_test)
# print(clf.predict_proba(features_test))
print("F1 Score:", f1_score(labels_test, predict_dt, average='weighted'))

F1 Score: 0.8309031208123631


K-Neighbors F1 Score: 0.7021963822807591

Gaussian Naive Bayes F1 Score: 0.9661538745149367

Decision Tree F1 Score: F1 Score: 0.8309031208123631